server/src/test/java/org/elasticsearch/action/admin/cluster/settings/SettingsUpdaterTests.java

```diff
@@ -471,4 +449,108 @@ public void testRemovingArchivedSettingsDoesNotRemoveNonArchivedInvalidOrUnknown
+        }
+        return unknownSettings;
+    }
+
+    private static List<Setting<String>> createInvalidSettings(int numberOfInvalidSettings) {
+        final List<Setting<String>> invalidSettings = new ArrayList<>(numberOfInvalidSettings);
+        for (int i = 0; i < numberOfInvalidSettings; i++) {
+            final Setting<String> invalidSetting = createInvalidSetting(i);
+            invalidSettings.add(invalidSetting);
+        }
+        return invalidSettings;
+    }
+
+    private static Setting<String> createInvalidSetting(int index) {
+        return Setting.simpleStringWithValidator("invalid.setting" + index,
+            new Setting.Validator<String>() {
+                @Override
+                public void validate(String value) {
+                    throw new IllegalArgumentException("invalid");
+                }
+
+                @Override
+                public void validate(String value, Map<Setting<String>, String> settings) {
+                    throw new IllegalArgumentException("invalid");
+                }
+            },
+            Property.NodeScope);
+    }
+
+    private static class FooLowSettingValidator implements Setting.Validator<Integer> {
+        @Override
+        public void validate(Integer value) {
+        }
+
+        @Override
+        public void validate(Integer low, Map<Setting<Integer>, Integer> settings) {
+            if (settings.containsKey(SETTING_FOO_HIGH) && low > settings.get(SETTING_FOO_HIGH)) {
+                throw new IllegalArgumentException("[low]=" + low + " is higher than [high]=" + settings.get(SETTING_FOO_HIGH));
+            }
+        }
+
+        @Override
+        public Iterator<Setting<Integer>> settings() {
+            return asList(SETTING_FOO_LOW, SETTING_FOO_HIGH).iterator();
+        }
+    }
+
+    private static class FooHighSettingValidator implements Setting.Validator<Integer> {
+        @Override
+        public void validate(Integer value) {
+        }
+
+        @Override
+        public void validate(Integer high, Map<Setting<Integer>, Integer> settings) {
+            if (settings.containsKey(SETTING_FOO_LOW) && high < settings.get(SETTING_FOO_LOW)) {
+                throw new IllegalArgumentException("[high]=" + high + " is lower than [low]=" + settings.get(SETTING_FOO_LOW));
+            }
+        }
+
+        @Override
+        public Iterator<Setting<Integer>> settings() {
+            return asList(SETTING_FOO_LOW, SETTING_FOO_HIGH).iterator();
+        }
+    }
+
+    private static final Setting<Integer> SETTING_FOO_LOW = new Setting<>("foo.low", "10",
+        Integer::valueOf, new FooLowSettingValidator(), Property.Dynamic, Setting.Property.NodeScope);
+    private static final Setting<Integer> SETTING_FOO_HIGH = new Setting<>("foo.high", "100",
+        Integer::valueOf, new FooHighSettingValidator(), Property.Dynamic, Setting.Property.NodeScope);
+
+    public void testUpdateOfValidationDependantSettings() {
```

Comment:
```suggestion
    public void testUpdateOfValidationDependentSettings() {
```

Just spotted a typo in the test name.